ibis/expr/api.py

```diff
@@ -1428,13 +1430,25 @@ def _string_like(self, pattern):
 
     Parameters
     ----------
-    pattern : string
+    pattern : str or List[str]
+        A pattern or list of patterns to match. If `pattern` is a list, then if
+        **any** pattern matches the input then the corresponding row in the
+        output is ``True``.
 
     Returns
     -------
-    matched : boolean value
+    matched : ir.BooleanColumn
     """
-    return _ops.StringSQLLike(self, pattern).to_expr()
+    return functools.reduce(
+        operator.or_,
+        map(
+            toolz.compose(
+                operator.methodcaller('to_expr'),
+                functools.partial(_ops.StringSQLLike, self),
+            ),
+            util.promote_list(patterns)
+        )
+    )
```

Comment:
Can you replace the fanciness from the map on down with a more vanilla list comprehension?